src/backends/postgresql/statement.cpp

```diff
@@ -781,19 +781,44 @@ void postgresql_statement_backend::describe_column(int colNum, data_type & type,
 
     default:
     {
-        int form = PQfformat(result_, pos);
-        int size = PQfsize(result_, pos);
-        if (form == 0 && size == -1)
+        auto typeCategoryIt = categoryByColumnOID_.find(typeOid);
+        if( typeCategoryIt == categoryByColumnOID_.end() ) 
         {
-            type = dt_string;
+            std::stringstream query;
+            query << "SELECT typcategory FROM pg_type WHERE oid=" << (int)typeOid;
+
+            PGresult* res = PQexec(session_.conn_, query.str().c_str());
+            if (PQresultStatus(res) != PGRES_TUPLES_OK) {
+                std::stringstream message;
+                message << "unknown data type with typelem: " << (int)typeOid
+                    << " for colNum: " << colNum
+                    << " with name: " << PQfname(result_, pos);
+                throw soci_error(message.str());
```

Comment:
It would be nice if we could extract this into a separate function (or even lambda if you prefer) to avoid duplicating almost the same code below. We could pass `\0` as category to indicate that it's unknown here.